docs/code/IDataViewDesignPrinciples.md

```diff
@@ -235,57 +234,91 @@ with a key type.
 ## Components
 
 The IDataView system includes several standard kinds of components and the
-ability to compose them to produce efficient data pipelines. A loader
-represents a data source as an `IDataView`. A transform is applied to an
-`IDataView` to produce a derived `IDataView`. A saver serializes the data
-produced by an `IDataView` to a stream, in some cases in a format that can be
-read by a loader. There are other more specific kinds of components defined
-and used by the ML.NET code base, for example, scorers, evaluators, joins, and
-caches. While there are several standard kinds of components, the set of
-component kinds is open.
-
-### Transforms
-
-Transforms are a foundational kind of IDataView component. Transforms take an
-IDataView as input and produce an IDataView as output. Many transforms simply
-"add" one or more computed columns to their input schema. More precisely,
-their output schema includes all the columns of the input schema, plus some
-additional columns, whose values are computed from some of the input column
-values. It is common for an added column to have the same name as an input
-column, in which case, the added column hides the input column. Both the
-original column and new column are present in the output schema and available
-for downstream components (in particular, savers and diagnostic tools) to
-inspect. For example, a normalization transform may, for each slot of a
-vector-valued column named Features, apply an offset and scale factor and
-bundle the results in a new vector-valued column, also named Features. From
-the user's perspective (which is entirely based on column names), the Features
-column was "modified" by the transform, but the original values are available
-downstream via the hidden column.
-
-Some transforms require training, meaning that their precise behavior is
-determined automatically from some training data. For example, normalizers and
-dictionary-based mappers, such as the TermTransform, build their state from
-training data. Training occurs when the transform is instantiated from user-
-provided parameters. Typically, the transform behavior is later serialized.
-When deserialized, the transform is not retrained; its behavior is entirely
-determined by the serialized information.
+ability to compose them to produce efficient data pipelines:
+
+Estimators and transformers. The langauge is derived from a similar idiom in Spark 
+
+A data loader represents a data source as an `IDataView`. A transformer is
```

Comment:
``A transformer is
applied via its `Fit` method to an `IDataView` to produce a derived
`IDataView`.``

Should `Fit` be `Transform` here?